### PR TITLE
change: ETCM-7985 pass network parameter to sidechain-main-cli

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

New versions of `sidechain-main-cli` expect a new parameter `--network (mainnet | testnet)` to be passed to the commands that access the chain. This PR adds this parameter where necessary, using the network value stored in the chain config.

‼️ I had to pin the partner-chains-smart-contracts version to the commit that added the parameter, since there's no tagged version that contains it yet. Before merging this PR it would probably be good to have a tagged release of PCSC but if that's too much waiting, we can stick to the commit hash.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [x] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ This changelog is based on [Keep A Changelog](https://keepachangelog.com/en/1.1.
 * polkadot-sdk dependency updated to partnerchains-stable2407 (stable2407 is v1.15.0 in the previous scheme)
 * remove Relay docker build files
 * changed the inner type of `McBlockHash` from Vec to an array
+* ETCM-7985 - bumper `partner-chains-smart-contracts` version and updated the `partner-chains-cli`
```
Comment (vvviiimmm) [inline changelog.md:11]:
```suggestion
* ETCM-7985 - bumped `partner-chains-smart-contracts` version and updated the `partner-chains-cli`
```

File: flake.nix [hunk 2]
```diff
@@ -18,7 +18,7 @@
     services-flake.url = "github:tgunnoe/services-flake";
 
     # Sidechains deps
-    trustless-sidechain.url = "github:input-output-hk/trustless-sidechain/v6.0.0-rc3";
+    trustless-sidechain.url = "github:input-output-hk/partner-chains-smart-contracts/692e7d2444229962b922e98a3eb16345f7d06f4c";
```
Comment (vvviiimmm) [inline flake.nix:21]: i'd wait for a proper tag

File: partner-chains-cli/src/config.rs [hunk 3]
```diff
@@ -265,6 +266,12 @@ impl CardanoNetwork {
 	pub fn to_id(&self) -> u32 {
 		self.0
 	}
+	pub fn to_network_param(&self) -> String {
```
Comment (vvviiimmm) [inline partner-chains-cli/src/config.rs:269]: minor naming: netowrk_param here vs network_arg in the parser

File: flake.nix (hunk 2 above)
Comment (AmbientTea) [reply flake.nix:21]: Yeah, I will

File: partner-chains-cli/src/config.rs (hunk 3 above)
Comment (AmbientTea) [reply partner-chains-cli/src/config.rs:269]: renamed the parser, it doesn't provide the arg, just the cardano network value, anyway

File: flake.nix (hunk 2 above)
Comment (LGLO) [reply flake.nix:21]: I don't know if we can convince SETs to sign off smart-contracts w/o having partner-chains-cli tests. Don't shoot messenger. I know who is right.

**Review by LGLO — APPROVED** [review]
The code changes look good to me, but please don't merge with having smart-contracts dependency out of order. Now as you reported there are even some problems with hashing keys with cardano-cli

**Review by vvviiimmm — APPROVED** [review]

File: partner-chains-cli/src/prepare_configuration/prepare_chain_params.rs [hunk 4]
```diff
@@ -95,12 +93,51 @@ fn is_gov_auth_valid_prompt() -> String {
...
+	let Some(vkey) = json.cbor_hex.strip_prefix(CBOR_KEY_PREFIX) else {
+		return Err(anyhow!(
+			"Invalid cbor value of public key - missing prefix {CBOR_KEY_PREFIX}: {}",
+			json.cbor_hex
+		));
+	};
+
+	let bytes: [u8; 32] = hex::decode(vkey)
+		.map_err(|err| {
+			anyhow!("Invalid cbor value of public key - not valid hex: {}\n{err:?}", json.cbor_hex)
```
Comment (vvviiimmm) [inline partner-chains-cli/src/prepare_configuration/prepare_chain_params.rs:124]: does it print the error on the newline or just prints \n? remember having some issues with that

Comment (AmbientTea) [reply partner-chains-cli/src/prepare_configuration/prepare_chain_params.rs]: It prints a proper newline

**Pull request merged**